src/Integration.UnitTests/Binding/ProjectBinderFactoryTests.cs

```diff
@@ -0,0 +1,109 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO.Abstractions;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Integration.Binding;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Binding
+{
+    [TestClass]
+    public class ProjectBinderFactoryTests
+    {
+        private ProjectBinderFactory testSubject;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            testSubject = new ProjectBinderFactory(Mock.Of<IServiceProvider>(), Mock.Of<IFileSystem>());
+        }
+
+        [TestMethod]
+        public void Ctor_NullServiceProvider_ArgumentNullException()
+        {
+            Action act = () => new ProjectBinderFactory(null, Mock.Of<IFileSystem>());
+
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("serviceProvider");
+        }
+
+        [TestMethod]
+        public void Ctor_NullFileSystem_ArgumentNullException()
+        {
+            Action act = () => new ProjectBinderFactory(Mock.Of<IServiceProvider>(),null);
+
+            act.Should().Throw<ArgumentNullException>().And.ParamName.Should().Be("fileSystem");
+        }
+
+        [TestMethod]
+        public void Get_CSharpProject_RoslynProjectBinderReturned()
+        {
+            var project = new ProjectMock("c:\\foo.proj");
+            project.SetCSProjectKind();
+
+            using (new AssertIgnoreScope())
```

Comment:
This is due to the fact that RoslynProjectBinder and CFamilyProjectBinder have inner dependencies that are being checked for not null in their CTORs. However, I don't want the factory class to know about them and I only need to verify that the correct type is being initialized.